optuna/integration/mlflow.py

```diff
@@ -102,56 +141,122 @@ def __init__(
-            study_direction = study.direction
-            if isinstance(study_direction, StudyDirection):
-                tags["direction"] = str(study_direction).split(".")[-1]
-
-            tags.update(trial.user_attrs)
-            distributions = {
-                (k + "_distribution"): str(v) for (k, v) in trial.distributions.items()
-            }
-            tags.update(distributions)
-
-            if self._tag_study_user_attrs:
-                tags.update(study.user_attrs)
-
-            # This is a temporary fix on Optuna side. It avoids an error with user
-            # attributes that are too long. It should be fixed on MLflow side later.
-            # When it is fixed on MLflow side this codeblock can be removed.
-            # see https://github.com/optuna/optuna/issues/1340
-            # see https://github.com/mlflow/mlflow/issues/2931
-            max_mlflow_tag_length = 5000
-            for key, value in tags.items():
-                value = str(value)  # make sure it is a string
-                if len(value) > max_mlflow_tag_length:
-                    tags[key] = textwrap.shorten(value, max_mlflow_tag_length)
-
-            mlflow.set_tags(tags)
+            self._set_tags(trial, study)
+
+    @experimental("2.9.0")
+    def track_in_mlflow(self) -> Callable:
+        """Decorator for using MLFlow logging in the objective function.
+
+        This decorator enables the extension of MLFlow logging provided by the callback.
+
+        All information logged in the decorated objective function will be added to the MLFlow
+        run for the trial created by the callback.
+
+        Returns:
+            ObjectiveFuncType: Objective function with tracking to MLFlow enabled
+        """
+
+        def decorator(func: ObjectiveFuncType) -> ObjectiveFuncType:
+            def wrapper(trial: optuna.trial.Trial) -> Union[float, Sequence[float]]:
+                study = trial.study
+                self._initialize_experiment(study)
+
+                with mlflow.start_run(run_name=str(trial.number), nested=self._nest_trials) as run:
+                    trial.set_system_attr(RUN_ID_ATTRIBUTE_KEY, run.info.run_id)
+
+                    return func(trial)
+
+            return wrapper
+
+        return decorator
+
+    def _initialize_experiment(self, study: optuna.study.Study) -> None:
+        """Initialize an MLFlow experiment with the study name.
+
+        If a tracking uri has been provided, MLFlow will be initialized to use it.
+
+        Args:
+            study: Study to be tracked in MLFlow
+        """
+        # This sets the tracking_uri for MLflow.
+        if self._tracking_uri is not None:
+            mlflow.set_tracking_uri(self._tracking_uri)
+
+        # This sets the experiment of MLflow.
+        mlflow.set_experiment(study.study_name)
+
+    def _set_tags(self, trial: optuna.trial.FrozenTrial, study: optuna.study.Study) -> None:
+        """Sets the Optuna tags for the current MLFlow run
```

Comment:
Should have a period.
```suggestion
        """Sets the Optuna tags for the current MLFlow run.
```